### PR TITLE
[Feat]: Export CSV file

### LINKED ISSUE #62: [Feat]: Export Function
### Describe

An export button to export the generated table.

### Describe alternatives you've considered

Export via csv or pdf.

### PR DESCRIPTION
**Pull request by jacky-p**
## Changes
_List Changes Introduced by this PR_
1. Added Export Button
2. HTML table to CSV file function
3. Added download functionality

## Purpose
Users can download a CSV file of the pairings of interns generated

## Approach
- Added `exportCSV.js` to scripts folder
- 3 functions, one to display the button, one to convert table to CSV, one to download the file

## TODOs
- [x] Add styling to the button
Screen Shot 2024-10-09 at 12 12 00 PM

Hover view:
Screen Shot 2024-10-09 at 12 12 11 PM

## Testing Steps
After generating intern pairs scroll to the bottom of the list and click download button, CSV file named `InternPairs.csv` should be downloaded which then can be opened/imported to Google sheets or Excel.

## Learning
- Followed this framework for creating a CSV from an HTML table since we organize our pairing information in a table.
  - [How to export HTML table to CSV using JavaScript ?](https://www.geeksforgeeks.org/how-to-export-html-table-to-csv-using-javascript/)
  - Worked with a lot of text modification to get the right format for the CSV
    - [String `.replace()` to replace new line `\n` with comma](https://www.w3schools.com/jsref/jsref_replace.asp)
    - [String `.split()` to separate text by the commas](https://www.geeksforgeeks.org/convert-comma-separated-string-to-array-using-javascrip)

Closes #62

### CODE REVIEW

File: src/scripts/exportCSV.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  let csv_data = [];
+
+  //add custom header to CSV
+  csv_data.push(
+    "Group, (Intern 1) Name, (Intern 1) Department, (Intern 1) Location, (Intern 2) Name, (Intern 2) Department, (Intern 2) Location",
+  );
+
+  // Get each row data by getting table and starting after header
+  let table = document.getElementById("interns-week-table");
+  let rows = table.getElementsByTagName("tr");
```
Comment (elimanzo) [inline src/scripts/exportCSV.js:35]: not being re-assigned so should be`const` also i think we can remove the intermediate variable
```suggestion
  const rows = document.getElementById("interns-week-table").getElementsByTagName("tr");
```

File: src/scripts/exportCSV.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+export function downloadCSVFile(csv_data) {
+  // Create CSV file object and feed our
+  // csv_data into it
+  let CSVFile = new Blob([csv_data], { type: "text/csv" });
+
+  // Create to temporary link to initiate
+  // download process
+  let temp_link = document.createElement("a");
```
Comment (elimanzo) [inline src/scripts/exportCSV.js:79]: objects are mutable so we can have a `const`

File: src/scripts/exportCSV.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+  // Download csv file
+  temp_link.download = "InternPairs.csv";
+  let url = window.URL.createObjectURL(CSVFile);
```
Comment (elimanzo) [inline src/scripts/exportCSV.js:83]: same as above :)

**Review by elimanzo — CHANGES_REQUESTED** [review]
Minor Nit and i think we're good to go!

File: src/scripts/exportCSV.js (hunk 3 above)
Comment (jacky-p) [reply src/scripts/exportCSV.js:83]: Resolved all the let statements that should be changed to const!

**Review by elimanzo — APPROVED** [review]
LGTM!

**Pull request merged**